src/System.Reflection/tests/PropertyInfo/PropertyInfo_GetSetValueTests.cs

```diff
@@ -0,0 +1,374 @@
+
+            Assert.Equal(strs2[0], strs[0]);
+        }
+
+        [Fact]
+        public static void TestGetSet_Name()
+        {
+            string propertyName = "Name";
+            InterfacePropertyImpl obj = new InterfacePropertyImpl();
+            PropertyInfo pi = GetProperty(typeof(InterfacePropertyImpl), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            var value = pi.GetValue(obj, (Object[])null);
+
+            Assert.Null(value);
+
+            //set value
+            string strs1 = "hello";
+
+            pi.SetValue(obj, strs1);
+            value = pi.GetValue(obj);
+            string strs2 = (string)value;
+            Assert.Equal(strs2, strs1);
+        }
+
+        [Fact]
+        public static void TestGet_PropertyC()
+        {
+            string propertyName = "PropertyC";
+            MyCoVariantTest obj = new MyCoVariantTest();
+            PropertyInfo pi = GetProperty(typeof(MyCoVariantTest), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            var value = pi.GetValue(obj, new Object[] { 1, "2" });
+
+            Assert.Null(value);
+        }
+
+        [Fact]
+        public static void TestGet_Property2()
+        {
+            string propertyName = "Property2";
+            LaterClass obj = new LaterClass();
+            PropertyInfo pi = GetProperty(typeof(LaterClass), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            int value = (int)pi.GetValue(obj);
+
+            Assert.Equal(value, 100);
+        }
+
+        //Try to set indexer Property
+        [Fact]
+        public static void TestSet_Item()
+        {
+            string propertyName = "Item";
+            TypeWithProperties obj = new TypeWithProperties();
+            PropertyInfo pi = GetProperty(typeof(TypeWithProperties), propertyName);
+            string[] h = { "hello" };
+
+            Assert.Equal(pi.Name, propertyName);
+
+            pi.SetValue(obj, "someotherstring", new Object[] { 99, 2, h, "f" });
+
+            Assert.Equal("992f1someotherstring", obj.setValue);
+
+            pi.SetValue(obj, "pw", new Object[] { 99, 2, h, "SOME  string" });
+
+            Assert.Equal("992SOME  string1pw", obj.setValue);
+        }
+
+        //
+        // Negative Tests for PropertyInfo
+        //
+
+        [Theory]
+        [InlineData("PropertyC", typeof(MyCoVariantTest), new Object[] { 1, "2", 3 })]
+        [InlineData("PropertyC", typeof(MyCoVariantTest), null)]
+        public static void TestGet_ThrowsTargetParameterCountException(string propertyName, Type type, Object[]testObj)
+        {
+            Object obj = Activator.CreateInstance(type);
+            PropertyInfo pi = GetProperty(type, propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            Assert.Throws<TargetParameterCountException>(() =>
+            {
+                var value = pi.GetValue(obj, testObj);
+            });
```

Comment:
Nit: the result here doesn't actually need to be stored into a local.  The xunit Assert.Throws method has overloads that accept Funcs instead of Actions, which then let you write code like:
```C#
Assert.Throws<TargetParameterCountException>(() => pi.GetValue(obj, testObj));
```
IMO, that's a bit cleaner, as it removes extra code that someone reading through the test doesn't need to focus on.